### PR TITLE
fix yolov5seg example

### PR DESCRIPTION
**Pull request by LihiStricher**

### CODE REVIEW

File: runtime/cpp/instance_segmentation/yolov5seg/README.md [hunk 1]
```diff
@@ -1,38 +1,73 @@
...
-Colors the detected objects pixels
-Prints the object detected + confidence to the screen
-Prints statistics
-NOTE: Currently supports only devices connected on a PCIe link.
-
-1. Dependencies:
-    - OpenCV, and g++-9:
-    ``` bash
+1. Install the requierments:
+    ```shell script
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/README.md:19]: requirements*

File: runtime/cpp/instance_segmentation/yolov5seg/README.md [hunk 2]
```diff
@@ -1,38 +1,73 @@
...
     sudo apt-get install -y libopencv-dev gcc-9 g++-9
     ```
-2. Build the project build.sh
-3. Run the executable:
-    ``` bash
-    ./build/segmentation_example_cpp -hef=HEF_PATH -path=VIDEO_PATH
+
+2. Install PyHailoRT
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/README.md:23]: pyhailort step is not relevant

File: runtime/cpp/instance_segmentation/yolov5seg/README.md [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
+        ```shell script
+        pip install
+        ```
+
+3. Download example files:
+    ```shell script
+    ./download_files.sh
+    ```
+
+5. Run the script:
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/README.md:38]: all of the steps besides #1 are not relevant

File: runtime/cpp/instance_segmentation/yolov5seg/README.md [hunk 4]
```diff
@@ -1,38 +1,73 @@
...
+Example 
+-------
+**Command**
+```shell script
+python supervision_example.py -i input_video.mp4 -o output_video.mp4 -l coco.txt-s 0.5
+```
+
+**Input**
+
+![Input example](./input.gif?raw=true)
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/README.md]: only output gif is enough, make sure to add the file

File: runtime/cpp/instance_segmentation/yolov5seg/get_hef_and_video.sh
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/get_hef_and_video.sh:1]: rename to download_resources.sh

File: runtime/cpp/instance_segmentation/yolov5seg/README.md [hunk 5]
```diff
@@ -1,38 +1,73 @@
...
-    ./build/segmentation_example_cpp -hef=yolov5n-seg.hef -path=full_mov_slow.mp4
 
-The output is a processed segmented video file - 'processed_video.mp4'
+Arguments
+---------
+
+- ``-m, --model``: Path to the pre-trained model file (HEF).
+- ``-i, --input``: Path to the input image or video on which object detection will be performed.
+
+Example
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/README.md]: incorrect

File: runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp [hunk 6]
```diff
@@ -79,15 +86,37 @@ Expected<std::shared_ptr<ConfiguredNetworkGroup>> configure_network_group(Device
     return std::move(network_groups->at(0));
 }
 
-template <typename T> hailo_status write_all(std::vector<InputVStream> &input, std::string &video_path,  
-                                            int height, int width, int channels, std::vector<cv::Mat>& frames) {
-    std::cout << "-I- Started write thread " << video_path << std::endl;
-    cv::VideoCapture capture(video_path);
+cv::Mat padding_frame(cv::Mat frame, int target_height, int target_width) {
```
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp:89]: padding_frame -> pad_frame

File: runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp
Comment (nina-vilela) [inline runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp:1]: convert_nms_with_byte_mask_buffer_to_detections could have a shorter name, maybe get_detections?

File: runtime/cpp/instance_segmentation/yolov5seg/README.md (hunk 3 above)
Comment (LihiStricher) [reply runtime/cpp/instance_segmentation/yolov5seg/README.md:38]: I updated this part in the second commit

**Pull request merged**